feat(File upload)

devvspaces opened the pull request:
- Created an endpoint for file uploads
- Endpoint returns the list of uploaded file URLs as response

![image](https://user-images.githubusercontent.com/77179231/205466909-4ea3467f-6d19-49b9-ac54-325f1b956a64.png)

Commit:
feat(File upload)

- Created an endpoint for file uploads

mikengr commented on backend/endpoints/files.py:
```diff
@@ -1,35 +1,76 @@
+router = APIRouter()
 
-# @router.get("/org/{org_id}/rooms/{room_id}/files")
-# async def get_files(org_id: str, room_id: str, page: int = 1, limit: int = 15):
```
@devvspaces is there a reason why the commented out endpoint was deleted?

mikengr commented on backend/config/settings.py:
```diff
@@ -2,6 +2,7 @@
 
 from pydantic import BaseSettings
 
+
```
@devvspaces please avoid touching files that are not related to the work you are doing. 8 files changes for this task gives me a lot of worry

mikengr commented on backend/endpoints/files.py:
```diff
@@ -1,35 +1,76 @@
+    if not isinstance(response, list):
+        raise HTTPException(
+            status_code=status.HTTP_424_FAILED_DEPENDENCY,
+            detail={"Message not sent": response},
```
this error checking is too ambiguous, not every error will be due to a failed dependency.

mikengr commented on backend/endpoints/files.py:
```diff
@@ -1,35 +1,76 @@
+    return JSONResponse(
+        content=ResponseModel.success(
+            data=message.dict(), message="Files uploaded successfully"),
+        status_code=status.HTTP_201_CREATED,
```
but no message was created here, you just returned the empty message object, how will the frontend implement this?

devvspaces replied:
@mikengr sir, they wanted just an upload endpoint, they said that's the way they wanted it, I had to do it for them.

mikengr reviewed (changes requested):
good work @devvspaces 
Please remove these `/other.png` and `/files.txt`, I can't understand their purpose in this PR
I have also left some comments on some of your changes.

Commit:
fix: upload files endpoint

- Made sure changes in  files that were not necessary to change were reverted.
- Updated the code to return specific errors for every error case.
- Made sure errors are well documented and made sure the error detail is easy to parse out
- Tested code in swagger and made sure it worked.
- Clients using this endpoint will have to use form data to send messages. More details about this in fast api [documentation](https://fastapi.tiangolo.com/tutorial/request-files/#:~:text=You%20can%20declare,the%20HTTP%20protocol.)
- Used an Header to get token from the client
- Token error handling where by if no files are uploaded, then token is not required.

Commit:
cleaned imports

Commit:
fix: upload files endpoint

- Made sure changes in files that were not meant to be changed are reverted
- Refactored the test for file storage's file upload method to pass after changes

Commit:
still trying to revert changes in those files

Commit:
still trying to revert changes in those files

devvspaces commented:
Test Screenshots Below ( Automated and Manual Tests) @mikengr 

## Automated tests and server startup screen shot

![test_1](https://user-images.githubusercontent.com/77179231/205575459-1b6ddd0a-fcac-4d2f-aec7-2c537453f968.png)

## API Endpoint doc

![test_2](https://user-images.githubusercontent.com/77179231/205575448-ff04193d-bc46-4aaa-ac45-55b8ccf79034.png)

## API Endpoint Manual Testing

![test_3](https://user-images.githubusercontent.com/77179231/205575463-a88dbade-5570-464f-9aee-faef9f5a2559.png)

mikengr reviewed (approved):
@devvspaces very good work, however putting your form validation in the same file as your endpoint is `anti-pattern`

In other to unblock others, I will merge this, but you will need to refactor `files.py`
